test/functional/rpc_signrawtransactionwithkey.py

```diff
@@ -126,10 +126,20 @@ def invalid_sighashtype_test(self):
         privkeys = [self.nodes[0].get_deterministic_priv_key().key]
         assert_raises_rpc_error(-8, "'all' is not a valid sighash parameter.", self.nodes[0].signrawtransactionwithkey, tx, privkeys, sighashtype="all")
 
+    def invalid_private_key_and_tx(self):
```

Comment:
nit:  An invalid private key and an invalid transaction could be tested separately rather than in the same function.  Was there a rationale for combining them?

Reply:
To avoid calling same stuff twice. e.g. `createrawtransaction`.